client/main_test.go

```diff
@@ -0,0 +1,59 @@
+package client_test
+
+import (
+	"flag"
+	"github.com/datastax/go-cassandra-native-protocol/compression/lz4"
+	"github.com/datastax/go-cassandra-native-protocol/compression/snappy"
+	"github.com/datastax/go-cassandra-native-protocol/frame"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	"os"
+	"testing"
+)
+
+func TestMain(m *testing.M) {
+	setLogLevel()
+	setRemoteAvailable()
+	createCodecs()
+	os.Exit(m.Run())
+}
+
+func setLogLevel() {
+	var logLevel int
+	flag.IntVar(
+		&logLevel,
+		"logLevel",
+		int(zerolog.InfoLevel),
+		"the log level to use (default: info)",
+	)
+	zerolog.SetGlobalLevel(zerolog.Level(logLevel))
```

Comment:
This is not working correctly because `flag.Parse` is not being called, it's always `INFO` no matter what parameter you pass. There should be a first phase where all the flags are defined then `flag.Parse` is called and then they can be used.